### PR TITLE
Version compatibility checks

### PR DESCRIPTION
**Pull request by achilleas-k**
Counterpart to G-Node/nix#638.

Compares supported file version with the format version of the file being opened and decides whether it can be opened in the requested mode (read only or read write).

Follows the [NIX documentation on versioning](https://github.com/G-Node/nix/blob/master/docs/versioning.md).

### CODE REVIEW

File: nixio/pycore/file.py [hunk 1]
```diff
@@ -25,6 +25,32 @@
     CFile = None
 
 
+FILE_FORMAT = "nix"
+HDF_FF_VERSION = (1, 0, 0)
```
Comment (gicmo) [inline nixio/pycore/file.py]: Should this not be `(1, 1, 0)`?

File: nixio/pycore/file.py [hunk 2]
```diff
@@ -90,16 +113,22 @@ def _open(cls, path, mode=FileMode.ReadWrite):
             path = path.encode("utf-8")
         except (UnicodeError, LookupError):
             pass
-        if not os.path.exists(path):
-            mode = FileMode.Overwrite
 
-        h5mode = map_file_mode(mode)
+        if (not os.path.exists(path) and mode == FileMode.ReadOnly):
```
Comment (gicmo) [inline nixio/pycore/file.py]: I don't think we need to outer `( )`, right?

File: nixio/pycore/file.py [hunk 3]
```diff
@@ -90,16 +113,22 @@ def _open(cls, path, mode=FileMode.ReadWrite):
...
+            raise RuntimeError(
+                "Cannot open non-existent file in ReadOnly mode!"
+            )
 
-        if os.path.exists(path):
-            newfile = cls._open_existing(path, h5mode)
-        else:
+        if (not os.path.exists(path)) or (mode == FileMode.Overwrite):
```
Comment (gicmo) [inline nixio/pycore/file.py]: In line 117 you don't out `()` around the inner conditions.

File: nixio/pycore/file.py (hunk 1 above)
Comment (achilleas-k) [inline nixio/pycore/file.py]: Yes, but not until it's changed in NIX master as well.

**Pull request merged**